src/utils/songInfo.ts

```diff
@@ -15,3 +15,102 @@ export const getPlaybackNotes = (
     .filter(track => track.channel === playbackChannel)
     .flatMap(track => track.notes);
 };
+
+type SmartMapping = number[];
+export type SmartMappingChangeEvent = {
+  time: number;
+  mapping: SmartMapping;
+};
+
+// TODO
+export const getSmartNotes = (
```

Comment:
So this will append a `smartKey` field to each note, so the waterfall can use this field to access the left margin